packages/my-react-package/src/MetaWallet.tsx

```diff
@@ -123,6 +127,42 @@ class MetaWallet {
         return newSignature;
     }
 
+    attachNonceProvider = (nonceProvider: string, api_key: string): void => {
+        this.nonceProviders[nonceProvider] = api_key;
+    }
+
+    getNonce = async (nonceProvider: string): Promise<string> => {
+        if (!this.nonceProviders[nonceProvider]) {
+            throw new Error("nonce Provider is not attached");
+        }
+
+        const nonce: string | null = localStorage.getItem('nonce');
+
+        if (nonce) return nonce;
+
+        const response = await axios.post(this.nonceProviders[nonceProvider],
+            {
+                webwallet_address: this.webWallet.address,
+            });
+
+
+        if (response.data && response.data.nonce !== 'Token expired') {
+
+            localStorage.setItem('nonce', response.data.nonce);
+            return response.data.nonce;
+        }
+
+        throw new Error("wallet is not authorized");
+    }
+
+    signNonce = async (nonce: string):
```

Comment:
Maybe we can name this function just `sign` so the user could use it for other purposes